vyper/functions/functions.py

```diff
@@ -1224,6 +1227,61 @@ def _can_compare_with_uint256(operand):
     )
 
 
+@signature('decimal')
+def sqrt(expr, args, kwargs, context):
+    from vyper.functions.utils import (
+        generate_inline_function,
+    )
+    arg = args[0]
+    sqrt_code = """
+assert x >= 0.0
+z: decimal = (x + 1.0) / 2.0
+y: decimal = x
+
+for i in range(256):
+    if (z >= y):
+        break
+    y = z
+    z = (x / z + z) / 2.0
+    """
+
+    x_type = BaseType('decimal')
+    placeholder_copy = ['pass']
+    # Steal current position if variable is already allocated.
+    if arg.value == 'mload':
```

Comment:
Does this still work if `arg.args[0]` is not a literal int?

Reply:
I guess it should .. I have my reservations about this pattern in general though because the frontend does not check that `x` does not get overwritten. I like the idea of always using `FREE_VAR_SPACE` because it's cheap and easy to reason about :)

Reply:
ha I know a way to fix that ;)

Reply:
Ah, missed that one. Nice!